laurent22/joplin
# Desktop: Fixes #10946: Stop crashing HTML/MD importer when content has link with very long name

Fixes https://github.com/laurent22/joplin/issues/10946

Related to OneNote Importer PR https://github.com/laurent22/joplin/pull/10642

# Summary

When working on OneNote PR I caught a bug where if a HTML had a link to a local resource with a very long name (>255 characters) `fsDriver-node` `stat` would throw an `ENAMETOOLONG`.

Initially my solution was to ignore certain link types (like `mailto:`), but while this can help ignore certain cases, I think th best solution is to treat this error similar to what already happens with `ENOENT`

I'm not sure if this change should also be implemented on others `fsDrivers`, so I just implemented where this seems to be a issue.

# Testing

I added an automated test.

It is also possible to test manually:

1. Create a HTML from the example HTML bellow
2. Import as a file
3. It should not crash

<details><summary>Example HTML:</summary>

```html
<html>
<body>
    <a href="1234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678123456781234567812345678.pdf" />
</body>
</html>
``` 

</details>
